netcontrold/lib/util.py

```diff
@@ -63,10 +63,17 @@ def cpuinfo():
         regex = re.compile('^(.*?)\s*:\s*(.*)')
         (param, val) = regex.match(line).groups()
 
+        if val=='':
+            continue
+
+        if param == 'processor' and val=='':
+            continue
+
         if param == 'processor':
             proc_list.append({})
 
-        proc_list[-1][param] = val
+        if(len(proc_list)>=1):
```

Comment:
Again, if proc_list is empty it was bug in parsing or somewhere else (may be /proc/cpuinfo not readable like that).. So raising exception is what this function could do, which the test has to assert.